src/sugar3/graphics/progressicon.py

```diff
@@ -14,86 +14,171 @@
+            '''
+       
         self._fill = get_surface(
             icon_name=icon_name, width=pixel_size, height=pixel_size,
             stroke_color=style.COLOR_TRANSPARENT.get_svg(),
             fill_color=fill_color)
-
-        self.connect("draw", self.__draw_cb)
+            '''
+            Sets fill color to transparent. 
+            '''
+        self.connect("draw", self.__draw_cb) '''Connect ---> Draw to self.__draw_cb'''
 
     def __draw_cb(self, widget, cr):
-        allocation = widget.get_allocation()
-
+     '''
+     Example:
+         def __draw_cb(self, da, cr): 
+         cr.set_source_rgb(1, 1, 1)
+         cr.paint()
+     '''
+        allocation = widget.get_allocation() 
+        
+        '''
+        Retruns:
+           A gtk.gdk.Rectangle
+           The get_allocation() method returns a rectangle containing the bounds of the widget's allocation.
+        '''
+        
         # Center the graphic in the allocated space.
-        margin_x = (allocation.width - self._stroke.get_width()) / 2
-        margin_y = (allocation.height - self._stroke.get_height()) / 2
+        margin_x = (allocation.width - self._stroke.get_width()) / 2 
+        margin_y = (allocation.height - self._stroke.get_height()) / 2 
         cr.translate(margin_x, margin_y)
 
         # Paint the fill, clipping it by the progress.
-        x_, y_ = 0, 0
-        width, height = self._stroke.get_width(), self._stroke.get_height()
-        if self._direction == 'vertical':  # vertical direction, bottom to top
+        x_, y_ = 0, 0 '''The value of margin x_ and y_ is 0,0 (0i,0j) '''
+        width, height = self._stroke.get_width(), self._stroke.get_height() '''Width = self._stroke.get_width() & height = self._stroke.get_height()'''
+        if self._direction == 'vertical':  #vertical direction, bottom to top 
             y_ = self._stroke.get_height()
             height *= self._progress * -1
-        else:
+        else: 
             rtl_direction = \
-                Gtk.Widget.get_default_direction() == Gtk.TextDirection.RTL
-            if rtl_direction:  # horizontal direction, right to left
-                x_ = self._stroke.get_width()
+                Gtk.Widget.get_default_direction() == Gtk.TextDirection.RTL '''Sets text direction from right to left'''
+            if rtl_direction: # horizontal direction, right to left
+                x_ = self._stroke.get_width() 
                 width *= self._progress * -1
-            else:  # horizontal direction, left to right
+            else: # horizontal direction, left to right
                 width *= self._progress
 
         cr.rectangle(x_, y_, width, height)
-        cr.clip()
+        cr.clip() 
         cr.set_source_surface(self._fill, 0, 0)
-        cr.paint()
+        cr.paint() 
 
         # Paint the stroke over the fill.
-        cr.reset_clip()
+        cr.reset_clip() 
         cr.set_source_surface(self._stroke, 0, 0)
         cr.paint()
```

Comment:
Please remove the documentation for this function for the same reason as stated above.